src/object.d

```diff
@@ -1703,7 +1703,7 @@ class TypeInfo_Class : TypeInfo
     override size_t getHash(scope const void* p) @trusted const
     {
         auto o = *cast(Object*)p;
-        return o ? o.toHash() : 0;
+        return hashOf(o ? o.toHash() : 0);
```

Comment:
If https://github.com/dlang/druntime/pull/2246 is accepted you will be able to revert this to `o ? o.toHash() : 0`.